BEP100.md

```diff
@@ -0,0 +1,154 @@
+
+BAS is not designed to have a BSC-compatible consensus or EVM execution environment, because BAS is technology-agnostic solution. Here we provide only basic functionality for the BAS applications and help them to set up their work and be a part of the BSC ecosystem. We can’t verify each operation on the chain and here we fully trust BAS developers, but we must be sure that validator transition is not compromised that is strictly required for the cross chain operations for the native asset. To reach this we’d like to introduce a block header verification function (BHVF) that must be specified by the BAS development team. BHVF is a function in the Solidity language that is able to verify block headers from BAS applications. This is a simplified version of the state verification function from Polkadot. BHVF is one of the required parameters to register BAS in the BSC’s smart contract. Also BHVF is responsible for verifying transaction receipt from blockchain that allows to prove correctness of cross chain transfer from BAS to BSC chain.
+
+Block header verification is not a very complicated function, but we will have to verify block headers to be sure that validator transition is well-done. Also using BLS/BN we don’t have to pass all block headers into it, it’s enough to publish only epoch blocks that contain new validator sets and signatures from all previous validators.
+
+If we assume that block verification consumes ~50k gas (w/ state modification), then we have next calculations:
++ per each block: 50k gas/block, then its ~16k gas/sec
++ per epoch: 50k gas/epoch (epoch can be from 5 minutes up to 1 day), then it's between ~160 gas/sec (for 5 min epoch) and 0.57 gas/sec (for 1 day epoch)
+
+Of course gas consumption depends on the epoch length and it's not required for BAS to have very short epochs. Epoch length of one day is recommended to be used for validator transition. It can be reduced based on application needs to 6-12 hours, but it doesn’t change gas consumption a lot.
+
+BAS brings native cross chain bridge and its embedded to BAS as a system smart contract. Here we specify interfaces for the EVM version of BAS:
+
+```solidity
+interface INativeAssetBridge {
+
+  function deposit() external payable;
+
+  function withdraw(bytes[] validatorSetSignatures, bytes transactionReceipt) external;
```

Comment:
Because `validatorSetSignatures` is present here,  to me, I think BHVF is redundant for the verification.  The source trust of BHVF is the block header, which gets only one signature from the validators, while in `withdraw`, we get the signatures from the majority validators.

Reply:
Just double confirm that the `validatorSetSignatures` here is the signature of blockheader?

Reply:
Agree, this is demo interfaces. Removing them from the doc.